oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/ClientRegistration.java

```diff
@@ -552,17 +572,25 @@ public Builder userInfoAuthenticationMethod(AuthenticationMethod userInfoAuthent
 		}
 
 		/**
-		 * Sets the attribute name used to access the user's name from the user info
-		 * response.
-		 * @param userNameAttributeName the attribute name used to access the user's name
-		 * from the user info response
-		 * @return the {@link Builder}
+		 * @deprecated Use {@link #usernameExpression(String)} instead
 		 */
+		@Deprecated
 		public Builder userNameAttributeName(String userNameAttributeName) {
```

Comment:
This is the place where I think you should default the usernameExpression to `"['" + userNameAttributeName + `"']"`.